src/metadataGeneration/typeResolver.ts

```diff
@@ -329,7 +333,7 @@ export class TypeResolver {
 
     if (typeNode.kind === ts.SyntaxKind.ArrayType) {
       const arrayType = typeNode as ts.ArrayTypeNode;
-      return this.getAnyTypeName(arrayType.elementType) + '[]';
+      return this.getAnyTypeName(arrayType.elementType) + 'Array';
```

Comment:
@WoH this is the change that I thought could be breaking for some users if they were relying on getting back `GenericRequestTypeAliasModel1[]` they would now get `GenericRequestTypeAliasModel1Array`

Reply:
I see. But I have more of these, if you're not sure about breaking changes, I think we can make it worth the 3.0 release :wink:

Reply:
@WoH would I be overthinking it if I added a swagger.config parameter called `noSymbolsInRefNames` that would allow @Asterix112 (and anyone else who has stricter needs for their ref names) to opt into this?

And then when we get to version 3.0 we can remove this `noSymbolsInRefNames`.

The reason why I'm resistant to going to a 3.0 version right now is (a) Asterix appears to need this soon and (b) there are a lot of great features that we can release without going to 3.0 (like your union PR, and the PR that fixes auto controller generation).

I appreciate any and all thoughts. Thank you for this collaboration @WoH. It's a lot of fun! :)

Reply:
[ ] in the $ref wasn't ever valid, was it?

Reply:
Wow, you are correct @WoH. It gives this error in SwaggerEditor: `ref values must be RFC3986-compliant percent-encoded URIs`.

So we're safe to make this change since it's a true bug fix.